eth-connector/src/lib.rs

```diff
@@ -556,31 +677,52 @@ impl FundsFinish for EthConnectorContract {
 
         log!("Finish deposit with the amount: {}", deposit_call.amount);
 
-        // Mint - calculate new balances
-        self.mint_eth_on_near(&deposit_call.new_owner_id, deposit_call.amount);
         // Store proof only after `mint` calculations
```

Comment:
This comment needs to be updated. 
Also, why don't we mint the whole amount to the `current_account_id` in the first place?

Reply:
We mint to the `current_account_id` just in case there is an `ft-transfer-call` when depositing to eth account, but on deposit to near account, we can mint directly to the `new_owner_id`.
If are asking how this worked with one call, then it worked because the `new_owner_id` == `current_account_id` on this `ft-transfer-call`.
additional to that it is not clear why actually the `env::predecessor_account_id(),` was used instead of `env::current_account_id()`

Reply:
10d1ab3ac41e26e55b890e87629fee46333d37aa